### PR TITLE
feat: added Messaging queue detail page

### PR DESCRIPTION
**Pull request by SagarRajput-7**
### Detail page 

- [x]  [Header title and page setup]
    - [x]  Title and page setup
    - [x]  Reset button and time picker
    - [x]  Options - (with other options as coming soon)
   
- [x]  [Config options] (using `auto-complete` API )
    - [x]  Consumer grp
    - [x]  Topic
    - [x]  Partitions

- [x]  [Main graph]
    - [x]  Time series graph
    - [x]  Tooltip
    - [x]  shareable link
    - [x]  Full screen view
    
- [x]  [Detail view]
    - [x]  Consumer grp
    - [x]  Producer details
    - [x]  Network latency
    - [x]  Redirections

#### Related Issues / PR's

https://github.com/SigNoz/engineering-pod/issues/1642

#### Screenshots

https://github.com/user-attachments/assets/69034ba6-fcab-4e31-adc4-e73f657d68ce

### CODE REVIEW

File: frontend/src/pages/MessagingQueues/MQCommon/MQCommon.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	return (
+		<Tooltip
+			title="Contact us at cloud-support@signoz.io for more details."
+			placement="top"
+			overlayClassName="tooltip-overlay"
+		>
+			<div className="coming-soon">
+				<div className="coming-soon__text">Coming Soon</div>
+				<div className="coming-soon__icon">
+					<Info size={10} color="var(--bg-sienna-400)" />
```
Comment (vikrantgupta25) [inline frontend/src/pages/MessagingQueues/MQCommon/MQCommon.tsx:17]: use `Color.BG_SIENNA_400`

File: frontend/src/pages/MessagingQueues/MQCommon/MQCommon.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+				</div>
+			</div>
+		</Tooltip>
+	);
+}
+
+export function SelectMaxTagPlaceholder(omittedValues: any[]): JSX.Element {
```
Comment (vikrantgupta25) [inline frontend/src/pages/MessagingQueues/MQCommon/MQCommon.tsx:24]: any ?

File: frontend/src/pages/MessagingQueues/MQDetails/MQTables/MQTables.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+} from 'pages/MessagingQueues/MessagingQueuesUtils';
+import { useEffect, useMemo, useState } from 'react';
+import { useMutation } from 'react-query';
+
+import {
+	ConsumerLagPayload,
+	getConsumerLagDetails,
+} from './getConsumerLagDetails';
+
+export function getColumns(data: any): any[] {
```
Comment (vikrantgupta25) [inline frontend/src/pages/MessagingQueues/MQDetails/MQTables/MQTables.tsx:25]: any ?

**Review by vikrantgupta25 — DISMISSED** [review]

File: frontend/src/pages/MessagingQueues/MQGraph/MQGraph.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	const widgetData = useMemo(
+		() => getWidgetQueryBuilder(getWidgetQuery({ filterItems })),
+		[filterItems],
+	);
+	const history = useHistory();
+	const location = useLocation();
+
+	return (
+		<Card
+			isDarkMode={isDarkMode}
```
Comment (vikrantgupta25) [inline frontend/src/pages/MessagingQueues/MQGraph/MQGraph.tsx:40]: `$isDarkMode`

File: frontend/src/pages/MessagingQueues/MQGraph/useGetAllConfigOptions.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+			setFetching(false);
+		}
+	};
+
+	useEffect(() => {
+		getValues();
+	}, [searchText]);
+
+	return { options: results, isFetching };
+}
```
Comment (ahmadshaheer) [inline frontend/src/pages/MessagingQueues/MQGraph/useGetAllConfigOptions.ts:59]: should we use react query's useQuery for fetching?

**Review by ahmadshaheer — APPROVED** [review]

File: frontend/src/pages/MessagingQueues/MQGraph/useGetAllConfigOptions.ts (hunk 5 above)
Comment (vikrantgupta25) [reply frontend/src/pages/MessagingQueues/MQGraph/useGetAllConfigOptions.ts:59]: +1

**Review by rahulkeswani101 — DISMISSED** [review]

File: frontend/src/pages/MessagingQueues/MQCommon/MQCommon.tsx (hunk 1 above)
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MQCommon/MQCommon.tsx:17]: done

File: frontend/src/pages/MessagingQueues/MQDetails/MQTables/MQTables.tsx (hunk 3 above)
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MQDetails/MQTables/MQTables.tsx:25]: done - under this `https://github.com/SigNoz/signoz/pull/5785/commits/cee05824138f1dc1b027d5658fbbca3419ebf928`

File: frontend/src/pages/MessagingQueues/MQCommon/MQCommon.tsx (hunk 2 above)
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MQCommon/MQCommon.tsx:24]: done

File: frontend/src/pages/MessagingQueues/MessagingQueuesUtils.ts [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+): void {
+	const selectedTimelineQuery: SelectedTimelineQuery = {
+		group: data?.group,
+		partition: data?.partition,
+		topic: data?.topic,
+		...getStartAndEndTimesInMilliseconds(timestamp),
+	};
+	urlQuery.set(
+		QueryParams.selectedTimelineQuery,
+		JSON.stringify(selectedTimelineQuery),
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MessagingQueuesUtils.ts:199]: done

File: frontend/src/pages/MessagingQueues/MQGraph/useGetAllConfigOptions.ts (hunk 5 above)
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MQGraph/useGetAllConfigOptions.ts:59]: done

**Review by rahulkeswani101 — DISMISSED** [review]

**Review by vikrantgupta25 — APPROVED** [review]
LGTM 🚀

**Review by YounixM — APPROVED** [review]

**Comment by YounixM** [comment]
@SagarRajput-7 : Please check the failed checks

**Review by ahmadshaheer — APPROVED** [review]

**Comment by SagarRajput-7** [comment]
> @SagarRajput-7 : Please check the failed checks

`commitlint / lint-commits` is failing due to the sub-PRs merged into this one, it shouldn't be an issue

**Pull request merged**